integration/commands_administration_test.go

```diff
@@ -299,126 +298,186 @@ func TestCommandsAdministrationGetParameter(t *testing.T) {
-			response: bson.D{
-				{"db", collection.Database().Name()},
-				{"collections", int32(1)},
-				{"views", int32(0)},
-				{"objects", int32(43)},
-				{"avgObjSize", 433.0},
-				{"dataSize", 8.192},
-				{"indexes", int32(0)},
-				{"indexSize", float64(0)},
-				{"totalSize", 16.384},
-				{"scaleFactor", float64(1_000)},
-				{"ok", float64(1)},
-			},
-		},
-		"ServerStatus": {
-			command: bson.D{{"serverStatus", int32(1)}},
-			response: bson.D{
-				{"host", ""},
-				{"version", "5.0.42"},
-				{"process", "handlers.test"},
-				{"pid", int64(0)},
-				{"uptime", int64(0)},
-				{"uptimeMillis", int64(0)},
-				{"uptimeEstimate", int64(0)},
-				{"localTime", primitive.DateTime(time.Now().Unix())},
-				{"catalogStats", bson.D{
-					{"collections", int32(1)},
-					{"capped", int32(0)},
-					{"timeseries", int32(0)},
-					{"views", int32(0)},
-					{"internalCollections", int32(0)},
-					{"internalViews", int32(0)},
-				}},
-				{"freeMonitoring", bson.D{{"state", "disabled"}}},
-				{"ok", float64(1)},
-			},
-		},
-	} {
-		name, tc := name, tc
-		t.Run(name, func(t *testing.T) {
-			t.Parallel()
+	var actual bson.D
+	command := bson.D{{"buildInfo", int32(1)}}
+	err := collection.Database().RunCommand(ctx, command).Decode(&actual)
+	require.NoError(t, err)
 
-			var actual bson.D
-			err := collection.Database().RunCommand(ctx, tc.command).Decode(&actual)
-			require.NoError(t, err)
+	doc := ConvertDocument(t, actual)
 
-			AssertEqualDocuments(t, tc.response, actual)
-		})
-	}
+	assert.Equal(t, float64(1), must.NotFail(doc.Get("ok")))
+	assert.Regexp(t, `^5\.0\.`, must.NotFail(doc.Get("version")))
+	assert.NotEmpty(t, must.NotFail(doc.Get("gitVersion")))
+
+	_, ok := must.NotFail(doc.Get("modules")).(*types.Array)
+	assert.True(t, ok)
+
+	assert.Equal(t, "deprecated", must.NotFail(doc.Get("sysInfo")))
+
+	versionArray, ok := must.NotFail(doc.Get("versionArray")).(*types.Array)
+	assert.True(t, ok)
+	assert.Equal(t, int32(5), must.NotFail(versionArray.Get(0)))
+	assert.Equal(t, int32(0), must.NotFail(versionArray.Get(1)))
+
+	assert.Equal(t, int32(strconv.IntSize), must.NotFail(doc.Get("bits")))
+	assert.False(t, must.NotFail(doc.Get("debug")).(bool))
+
+	assert.Equal(t, int32(16777216), must.NotFail(doc.Get("maxBsonObjectSize")))
+	_, ok = must.NotFail(doc.Get("buildEnvironment")).(*types.Document)
+	assert.True(t, ok)
+}
+
+func TestCommandsAdministrationCollStats(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup(t)
+
+	var actual bson.D
+	command := bson.D{{"collStats", collection.Name()}}
+	err := collection.Database().RunCommand(ctx, command).Decode(&actual)
+	require.NoError(t, err)
+
+	doc := ConvertDocument(t, actual)
+	assert.Equal(t, float64(1), must.NotFail(doc.Get("ok")))
+	assert.Equal(t, collection.Database().Name()+"."+collection.Name(), must.NotFail(doc.Get("ns")))
+	assert.LessOrEqual(t, int32(0), must.NotFail(doc.Get("count")))
+	assert.LessOrEqual(t, int32(0), must.NotFail(doc.Get("size")))
```

Comment:
Why do all those things use `LessOrEqual`? Those values could be less than 0?

Reply:
> Why do all those things use `LessOrEqual`? Those values could be less than 0?

It's about `count` value to be bigger or equal to 0.

Reply:
@seeforschauer `LessOrEqual` is about `count` value to be bigger or equal to 0? Sound quite hard to understand (why not `GreaterOrEqual`?) And also, why can't we just use `Equal` if we expect an exact 0 here?

Reply:
> @seeforschauer `LessOrEqual` is about `count` value to be bigger or equal to 0? 

LessOrEqual asserts that the first element is less than or equal to the second

> Sound quite hard to understand (why not `GreaterOrEqual`?) 

probably. But if you put the variables in order, then it's more easier: `0 <= value`. Value has to be bigger.

> And also, why can't we just use `Equal` if we expect an exact 0 here?
I consider adding `LessOrEqual` everywhere as I and @AlekSi agreed on the call that the later implementation of the entire command will be later on for each separate command. 
So in this PR, I'm about to add some checks only, for the variables where I can - add some more checks. For example If I expect an array, check for an array, If I expect IP, check IP regexp. If I expect a `5.0` Mongo version and I can skip minor version, I check for 5.0. 


These tests are fluky.  Let's discuss it as we don't want to do fluky tests lightly.

Reply:
> These tests are fluky. Let's discuss it as we don't want to do fluky tests lightly.

So that's my suggestion - to add `LessOrEqual` where it's necessary.